### PR TITLE
Add contributing documentation

### PR DESCRIPTION
**Pull request by rmccue**
Hoping this unblocks the process from relying on me and/or Joe.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Relaxing rules can be done in minor releases, but generally should be done in major releases if it's a major change (for example, allowing different file names). Use your best judgement to decide what is a major and what is a minor change, and if in doubt, run it past @joehoyle or @rmccue.
+
+Generally, so long as changes to rules have consensus, they are fine to be published. Any controversial rules should be widely discussed, and if a tie-breaker is needed, @joehoyle can make a final call. If you're not sure, ask @rmccue. Non-controversial changes or bugfixes do not need input from @joehoyle or @rmccue provided versioning and release processes are all followed.
+
+
+## Releasing
+
+Any changes which cause existing, working production code to fail should trigger a new major release. Only bugfixes or making rules more lenient should be in minor releases.
+
+When publishing major releases, these need to be published in a two-step process. First, publish the standards, then bump the defaults after some time. This gives projects time to assess the changes and migrate at their own pace.
```
Comment (mikeselander) [inline CONTRIBUTING.md]: Could you give an idea of how much time you'd expect for this period? Perhaps two weeks would be sufficient?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+The process for releasing is:
+
+* Ensure your working directory is clean and up-to-date on `master`
+* Run `yarn publish` and enter the tag number
+	* This publishes to npm, commits the version change, and creates a corresponding git tag
+* Push the version bump commit to `master`: `git push`
+* Push the new tag: `git push --tags`
+	* This triggers Packagist to release the new version
+* Run `./publish.sh` to push the standards for hm-linter
+	* This will ask if you want to bump the latest version to the new version. Only do this for minor releases.
```
Comment (mikeselander) [inline CONTRIBUTING.md]: We should probably also add writing up release notes for our `dev` P2 and adding changelog notes on GitHub here.

**Comment by rmccue** [comment]
@mikeselander Addressed your feedback here. For the time between release and default bump for major releases, generally 2+ sprints: assume you release after a sprint has already been scoped, so earliest the work can happen is in the next sprint. Even bigger changes (say, where you're trying to shift existing behaviour by making rules stricter) may need longer as the changes to the codebase may be larger.

**Comment by rmccue** [comment]
Unsure why tests are failing here, is `master` broken?

**Comment by mikeselander** [comment]
Looks like it, I'm going to take a look this morning. I wouldn't have expected your PR to be failing though 🤔

**Pull request merged**